### ISSUE TITLE
[Feedback] cool things to do

### ISSUE THREAD

**Issue by mlnomadpy** [ridaouiafaf/heart_disease_uci_prediction#1]
sup just saw the project on mlnomads
cool stuff
down below is just some more cool stuff to add if you are interested in exploring how to productionize a bit your script

- first thing first, try to always use functional  programming more, even thought it is easy to just run the code in a cell with jupyter notebooks,  but having functions will help you re-use codes in different other notebooks, say you  want to  create two separate runs, one for training and one for evals, instead of having the same code block for both scripts, you can have a function that jsut do that, and import it in both, now if you wanna add some improvment to the function, it will propagate to both your train run and your eval run

- which brings me to the second point, i hate jupyter, and most of the ops do too, you can't automate stuff easily, ops is hell, and there is too much problems once you want to go  to production with your script, use modular code, a file for each function (if you wanna go hardcore), code once reuse every where, you won't have problem with hidden states, eacch time you run the script new process is created.
- wandb is your friend for logging, while having everything in a notebook is easy, to compare the different runs of each notebook is a whole new level of hell, why suffer in  a notebook when you can live happy in wandb?
- data is not static in the real world, it moves, so your system need to move with it, and you don't want to have a job where you are just re run the same scripts manually, so you will need to automate stuff like statistics analysis, while it is necessary for humans to visualize data, for code, it needs threshholds to know when is my knew model better than the other, how much drift/skew is my new data coming from production, is the new model better than the deployed one, all this could be automated (tfx is cool for this and simple  to use)
- use git to track your changes, and deploy pipelines (most of the work you will do on the training server is git pull, python main.py --please-work) and then re itrate

- and if you wanna still be friends don't use this model.fit(X_train,y_train)  it is the easiest path to limiting your progress in AI, while it is easy, you will be limted to only calling it, while if you wanna use some other pre-training methods you won't find a script that gives you .fit , and you will need to learn how to implement training functions for normal gradient descent before you will be able to move to contrastive learning, self supervised learning, MLM.... which is the new paradigm of building ai model right now.

w hada makan,
wa salamoun 3ala lmoursalin
